editor/animation_track_editor.cpp

```diff
@@ -5297,6 +5297,28 @@ void AnimationTrackEditor::_add_track(int p_type) {
 		return;
 	}
 	adding_track_type = p_type;
+	Vector<StringName> valid_types;
+	switch (adding_track_type) {
+		case Animation::TYPE_BLEND_SHAPE: {
+			// Blend Shape is a property of MeshInstance3D
+			valid_types.push_back(StringName("MeshInstance3D"));
+		} break;
+		case Animation::TYPE_POSITION_3D:
+		case Animation::TYPE_ROTATION_3D:
+		case Animation::TYPE_SCALE_3D: {
+			// 3D Properties come from nodes inheriting Node3D
+			valid_types.push_back(StringName("Node3D"));
+		} break;
+		case Animation::TYPE_AUDIO: {
+			valid_types.push_back(StringName("AudioStreamPlayer"));
+			valid_types.push_back(StringName("AudioStreamPlayer2D"));
+			valid_types.push_back(StringName("AudioStreamPlayer3D"));
+		} break;
+		case Animation::TYPE_ANIMATION: {
+			valid_types.push_back(StringName("AnimationPlayer"));
```

Comment:
```suggestion
			// Blend Shape is a property of MeshInstance3D.
			valid_types.push_back(SNAME("MeshInstance3D"));
		} break;
		case Animation::TYPE_POSITION_3D:
		case Animation::TYPE_ROTATION_3D:
		case Animation::TYPE_SCALE_3D: {
			// 3D Properties come from nodes inheriting Node3D.
			valid_types.push_back(SNAME("Node3D"));
		} break;
		case Animation::TYPE_AUDIO: {
			valid_types.push_back(SNAME("AudioStreamPlayer"));
			valid_types.push_back(SNAME("AudioStreamPlayer2D"));
			valid_types.push_back(SNAME("AudioStreamPlayer3D"));
		} break;
		case Animation::TYPE_ANIMATION: {
			valid_types.push_back(SNAME("AnimationPlayer"));
```

Reply:
I've committed the change here. Is there a particular reasoning behind using SNAME vs StringName? or even SceneStringName? Or plain string with just quotes?

Reply:
It is a dedicated macro creating a static instance, otherwise this creates a new `StringName` instance each time this method is run, forgot to add that to the comment when remaking it after some other changes